lib/src/modules/task.cc

```diff
@@ -240,30 +241,54 @@ static std::string createUrlEndpoint(const lth_jc::JsonContainer& uri) {
 //    the same task file.
 // The downloaded task file's permissions will be set to rwx for user and rx for
 // group for non-Windows OSes.
-static void downloadTaskFile(const std::vector<std::string>& master_uris,
-                             lth_curl::client& client,
-                             const fs::path& file_path,
-                             const lth_jc::JsonContainer& uri) {
-    auto url = master_uris[0] + createUrlEndpoint(uri);
-    lth_curl::request req(url);
-    // timeout from connection after one minute, can configure
-    req.connection_timeout(60000);
+//
+// The method returns a tuple (success, err_msg). success is true if the file was downloaded;
+// false otherwise. err_msg contains the most recent http_file_download_exception's error
+// message; it is initially empty.
+static boost::tuple<bool, std::string> downloadTaskFile(const std::vector<std::string>& master_uris,
+                                                      lth_curl::client& client,
+                                                      const fs::path& file_path,
+                                                      const lth_jc::JsonContainer& uri) {
+    auto endpoint = createUrlEndpoint(uri);
+    boost::tuple<bool, std::string> result = boost::make_tuple(false, "");
```

Comment:
Use `auto result = std::make_tuple(false, std::string{})`.

Reply:
Unfortunately I can't, b/c auto makes the result a constant tuple so I can't modify it.